client/post-editor/editor-confirmation-sidebar/index.jsx

```diff
@@ -41,6 +41,20 @@ class EditorConfirmationSidebar extends Component {
 		status: PropTypes.string,
 	};
 
+	state = {
+		hasUnclosed: false,
+	};
+
+	static getDerivedStateFromProps( props ) {
+		// In order to improve performance, `hasUnclosed` determines whether the
+		// sidebar should be rendered or not. The content has to be rendered
+		// the first time the sidebar is not in a 'closed' status.
+		if ( props.status !== 'closed' ) {
+			return { hasUnclosed: true };
```

Comment:
Unimporant optimization: check for `! state.hasUnclosed && props.status !== 'closed'` and avoid an unneeded `setState` call.

Reply:
Good idea, Thanks!